stream_alert/rule_processor/rules_engine.py

```diff
@@ -172,6 +174,14 @@ def process_subkeys(cls, record, payload_type, rule):
             return True
 
         for key, nested_keys in rule.req_subkeys.iteritems():
+            # This check is neceesary with loose schemas when a key's
+            # value will likely contain multiple types.  An example of this
```

Comment:
Unless I'm misreading the code, you're `if` statement doesn't support any set of multiple types, it only supports the option of it being a a given type or non-truthy value